doc/cygnus-ngsi-ld/user_and_programmer_guide/adding_new_sink.md

```diff
@@ -0,0 +1,133 @@
+    * [Inherited starting and stoping](#section2.2)
+    * [Inherited events consumption](#section2.3)
+    * [Inherited counters ](#section2.4)
+* [New sink class](#section3)
+    * [Specific configuration](#section3.1)
+    * [Kind of information to be persisted](#section3.2)
+    * [Fitting to the specific data structures](#section3.3)
+* [Backend convenience classes](#section4)
+* [Naming and placing the new sink](#section5)
+
+## <a name="section1"></a>Introduction
+`cygnus-ngsi-ld` allows for NGSI-LD context data persistence in certain storages by means of Flume sinks. As long as the current collection of sinks could be limited for your purposes, you can add your own sinks regarding a persistence technology of your choice and become an official `cygnus-ngsi-ld` contributor!
+
+This document tries to guide you on the development of such alternative sinks, by giving you guidelines about how to write the sink code, but also how the different classes must be called, the backends that can be used, etc.
+
+[Top](#top)
+
+## <a name="section2"></a>Base `NGSILDSink` class
+`NGSILDSink` is the base class all the sinks within `cygnus-ngsi-ld` extend. It is an abstract class which extends from `CygnusSink` class at `cygnus-common` (which, by its side, extends Flume's native `AbstractSink`).
+
+`NGSILDSink` provides most of the logic required by any NGSI-like sink:
+
+* Configuration of parameters common to all the sinks.
+* Starting and stopping the sink.
+* Flume events consumption in a batch-like approach, including opening, committing and closing of Flume transactions.
+* Counters for statistics (in fact, this feature is given by `CygnusSink`).
+
+You find this class at the following path:
+
+    fiware-cygnus/cygnus-ngsi-ld/src/main/java/com/telefonica/iot/cygnus/sinks/NGSILDSink.java
+
+[Top](#top)
+
+### <a name="section2.1"></a>Inherited configuration
+All the sinks extending `NGSILDSink` inherit the following configuration parameters:
+
+| Parameter | Mandatory | Default value | Comments |
+|---|---|---|---|
+| batch_size | no | 1 | Number of events accumulated before persistence. |
+| batch_timeout | no | 30 | Number of seconds the batch will be building before it is persisted as it is. |
+| batch_ttl | no | 10 | Number of retries when a batch cannot be persisted. Use `0` for no retries, `-1` for infinite retries. Please, consider an infinite TTL (even a very large one) may consume all the sink's channel capacity very quickly. |
+| data_model | no | dm-by-entity | Accepted values: <i>dm-by-service</i>, <i>dm-by-service-path</i>, <i>dm-by-entity</i>, <i>dm-by-entity-type</i> and <dm-by-attribute</i>. |
+| enable_grouping | no | false | Accepted values: <i>true</i> or <i>false</i>. |
+| enable\_lowercase | no | false | Accepted values: <i>true</i> or <i>false</i>. |
+
+These parameters are read (and defaulted, when required) in the `configure(Context)` method.
+
+[Top](#top)
+
+### <a name="section2.2"></a>Inherited starting and stoping
+TBD
```

Comment:
Pending to be completed?

Reply:
Yes, we have not prepared this section yet. I think this is more related to the developers guide so we can update this in the future.

Reply:
I have checked that in the original documentation for cygnus-ngsi (https://github.com/telefonicaid/fiware-cygnus/blob/master/doc/cygnus-ngsi/user_and_programmer_guide/adding_new_sink.md#section2.2) this section is also in TBD status, so it can be leave also in this state for cygnus-ngsi-ld.

However, maybe the following change makes sense:

```suggestion
TBD (as in doc/cygnus-ngsi/user_and_programmer_guide/adding_new_sink.md)
```

to "link" both them.